-1

Tested at: 1d75bb9b0635cf143a352582917b892dabb7064c

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26066/33417/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation warning when building:  See details on the summary page.

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```


<pre>+ eval scram build outputlog '&&' '(/build/cmsbld/jenkins/workspace/ib-any-integration/cms-bot/buildLogAnalyzer.py' --logDir /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-04-2300/tmp/slc7_amd64_gcc700/cache/log/src '||' 'true)'
++ scram build outputlog
Reading cached build data
Entering library rule at src/SimGeneral/Debugging/plugins
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-04-2300/src/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-04-2300/src/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc:313:17: error: array index 2 is past the end of the array (which contains 2 elements) [-Werror,-Warray-bounds]
  if(bhgeomnew) hgtopo[2] = &(bhgeomnew->topology());
                ^      ~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-04-2300/src/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc:296:3: note: array 'hgtopo' declared here
  const HGCalTopology*     hgtopo[2];
  ^
</pre>

--- a/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc
+++ b/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc
@@ -294,7 +294,7 @@ void CaloParticleDebugger::fillSimHits(
   const HcalGeometry *bhgeom = nullptr;
   const HGCalDDDConstants* hgddd[3];
   const HGCalTopology*     hgtopo[3];
-  const HcalDDDRecConstants* hcddd;
+  const HcalDDDRecConstants* hcddd = nullptr;
   eegeom = static_cast<const HGCalGeometry*>(geom->getSubdetectorGeometry(DetId::HGCalEE,ForwardSubdetector::ForwardEmpty));
   //check if it's the new geometry
   if(eegeom){
@@ -332,6 +332,7 @@ void CaloParticleDebugger::fillSimHits(
         id = simId;
       }
       else if (isHcal) {
+        assert(hcddd);
         HcalDetId hid = HcalHitRelabeller::relabel(simId, hcddd);
         if (hid.subdet() == HcalEndcap) id = hid;
       } else {

--- a/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc
+++ b/SimGeneral/Debugging/plugins/CaloParticleDebugger.cc
@@ -292,8 +292,8 @@ void CaloParticleDebugger::fillSimHits(
   iSetup.get<CaloGeometryRecord>().get(geom);
   const HGCalGeometry *eegeom = nullptr, *fhgeom = nullptr, *bhgeomnew = nullptr;
   const HcalGeometry *bhgeom = nullptr;
-  const HGCalDDDConstants* hgddd[2];
-  const HGCalTopology*     hgtopo[2];
+  const HGCalDDDConstants* hgddd[3];
+  const HGCalTopology*     hgtopo[3];
   const HcalDDDRecConstants* hcddd;
   eegeom = static_cast<const HGCalGeometry*>(geom->getSubdetectorGeometry(DetId::HGCalEE,ForwardSubdetector::ForwardEmpty));
   //check if it's the new geometry

